spec/requests/media/odk_response_spec.rb

```diff
@@ -69,128 +69,176 @@
+
+        post_submission(submission_file2, "sassafras.jpg", image2, false)
+        expect_submission(:created, 1, 3)
+
+        post_submission(submission_file3, "sassafras.jpg", image2, false)
+        expect_submission(:created, 1, 3)
+
+        expect_answers
+      end
     end
-  end
 
-  context "with multiple parts" do
-    let(:form) { create(:form, :live, question_types: %w[text image sketch]) }
+    context "with multiple parts, duplicate first submission at the end" do
+      let(:form) { create(:form, :live, question_types: %w[text image sketch]) }
 
-    it "should successfully process the submission", database_cleaner: :truncate do
-      image = Rack::Test::UploadedFile.new(image_fixture("the_swing.jpg"), "image/jpeg")
-      image2 = Rack::Test::UploadedFile.new(image_fixture("the_swing.jpg"), "image/jpeg")
-      submission_file = prepare_and_upload_submission_file("multiple_part_media.xml")
-      submission_file2 = prepare_and_upload_submission_file("multiple_part_media.xml")
-
-      # Submit first part
-      post submission_path,
-        params: {
-          xml_submission_file: submission_file,
-          "the_swing.jpg" => image,
-          "*isIncomplete*" => "yes"
-        },
-        headers: auth_header
-      expect(response).to have_http_status(:created)
-      expect(Response.count).to eq(1)
+      it "should successfully process the submission", database_cleaner: :truncate do
+        image = Rack::Test::UploadedFile.new(image_fixture("the_swing.jpg"), "image/jpeg")
+        image2 = Rack::Test::UploadedFile.new(image_fixture("sassafras.jpg"), "image/jpeg")
+        submission_file = prepare_and_upload_submission_file("multiple_part_media.xml")
+        submission_file2 = prepare_and_upload_submission_file("multiple_part_media.xml")
+        submission_file3 = prepare_and_upload_submission_file("multiple_part_media.xml")
 
-      # Submit second part
-      post submission_path,
-        params: {
-          xml_submission_file: submission_file2,
-          "another_swing.jpg" => image2
-        },
-        headers: auth_header
+        post_submission(submission_file, "the_swing.jpg", image, true)
+        expect_submission(:created, 1, 2)
 
-      expect(response).to have_http_status(:created)
+        post_submission(submission_file2, "sassafras.jpg", image2, false)
+        expect_submission(:created, 1, 3)
 
-      form_response = Response.first
+        post_submission(submission_file, "the_swing.jpg", image, true)
+        expect_submission(:created, 1, 3)
 
-      expect(form_response.form).to eq(form)
-      expect(form_response.answers.count).to eq(3)
+        expect_answers
+      end
+    end
+
+    # Currently database unique constraint does not allow two responses with the same form_id and odk_xml checksum
+    # We should consider if we want to remove this constraint?
+    # What is the liklihood that two users will submit the exact same response and formid?
```

Comment:
if nobody's complained so far, it seems exceedingly unlikely and not worth spending more effort supporting it. feel free to get rid of these commented specs.

Reply:
done